packages/block-editor/src/components/block-list/style.scss

```diff
@@ -168,8 +168,28 @@
 			opacity: 1;
 		}
 	}
-}
 
+	// Add extra border to parent blocks when their children are selected.
+	&.has-child-selected {
+
+		> .block-editor-block-list__block-edit::before {
+			border: $border-width dashed $dark-opacity-light-800;
+		}
+
+		> .block-editor-block-list__block-edit > [data-block] > div > .block-editor-inner-blocks > .block-editor-block-list__layout > .block-editor-block-list__block:not(.is-selected) > .block-editor-block-list__block-edit::before,
+		> .block-editor-block-list__block-edit > [data-block] > div > .wp-block-cover__inner-container > .block-editor-inner-blocks > .block-editor-block-list__layout > .block-editor-block-list__block:not(.is-selected) > .block-editor-block-list__block-edit::before,
+		> .block-editor-block-list__block-edit > [data-block] > div > .wp-block-group__inner-container > .block-editor-inner-blocks > .block-editor-block-list__layout > .block-editor-block-list__block:not(.is-selected) > .block-editor-block-list__block-edit::before {
+			border: $border-width dashed $dark-opacity-light-800;
+		}
+	}
+
+	// Add extra border to child blocks when they are selected.
+	&.is-selected > .block-editor-block-list__block-edit > [data-block] > div > .block-editor-inner-blocks > .block-editor-block-list__layout > .block-editor-block-list__block:not(.is-selected) > .block-editor-block-list__block-edit::before,
+	&.is-selected > .block-editor-block-list__block-edit > [data-block] > div > .wp-block-cover__inner-container > .block-editor-inner-blocks > .block-editor-block-list__layout > .block-editor-block-list__block:not(.is-selected) > .block-editor-block-list__block-edit::before,
+	&.is-selected > .block-editor-block-list__block-edit > [data-block] > div > .wp-block-group__inner-container > .block-editor-inner-blocks > .block-editor-block-list__layout > .block-editor-block-list__block:not(.is-selected) > .block-editor-block-list__block-edit::before {
```

Comment:
Is this amount of specificity really needed? Quite apart from the performance impact, it'll be a nightmare if we ever need to override this for some reason.

Reply:
Unfortunately it is, for now. 😞 

Because of the many, many nested divs that make up a block, there's no other way to target just a "direct" child of the selected block without these sorts of selectors. This was a problem for the Group block too — you'll see these sorts of [highly-specific selectors there too](https://github.com/WordPress/gutenberg/blob/c80a0089cc3d24c4a01e693e2fc209279a74f45a/packages/block-library/src/group/editor.scss).

Reply:
Oh wow, OK.

Reply:
Could this high-specificity selectors be solved by new classNames: `is-direct-parent-selected`. Not a blocker but it's definitely something that can break pretty easily and we might need to think of a more robust approach.